src/blocking.rs

```diff
@@ -40,6 +42,24 @@ impl<E: fmt::Debug> fmt::Display for WriteFmtError<E> {
 #[cfg(feature = "std")]
 impl<E: fmt::Debug> std::error::Error for WriteFmtError<E> {}
 
+/// Error returned by [`Write::write_all`]
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum WriteZeroError<E> {
```

Comment:
I'd call this `WriteAllError`, named after the function that returns it, consistent with the other errors.